src/platform_linux.cc

```diff
@@ -154,13 +154,69 @@ std::string GetWorkingDirectory() {
   return working_dir;
 }
 
+// Returns the canonicalized absolute pathname, without expanding symbolic links.
```

Comment:
It'd be awesome if you could write some test for this, though I understand it may be tricky. I suspect this could be a source of bugs in the future. Maybe mock out the stat/etc calls?

Reply:
It is tricky to write a test because it calls `stat`. I'll have to make some assumption of the file hierarchy.